File: php/features.php
<?php
declare(strict_types=1);

// Bng2latlongConverter SDK feature factory

require_once __DIR__ . '/feature/BaseFeature.php';
require_once __DIR__ . '/feature/RatelimitFeature.php';
require_once __DIR__ . '/feature/RetryFeature.php';
require_once __DIR__ . '/feature/TestFeature.php';
require_once __DIR__ . '/feature/TimeoutFeature.php';


class Bng2latlongConverterFeatures
{
    public static function make_feature(string $name)
    {
        switch ($name) {
            case "base":
                return new Bng2latlongConverterBaseFeature();
            case "ratelimit":
                return new Bng2latlongConverterRatelimitFeature();
            case "retry":
                return new Bng2latlongConverterRetryFeature();
            case "test":
                return new Bng2latlongConverterTestFeature();
            case "timeout":
                return new Bng2latlongConverterTimeoutFeature();
            default:
                return new Bng2latlongConverterBaseFeature();
        }
    }

    /**
     * Does a generated feature class back this name? False for a name only
     * an options extend instance can supply (the station adopt path) - the
     * constructor uses this to skip make_feature for such names instead of
     * adding a stray BaseFeature.
     */
    public static function has_feature(string $name): bool
    {
        switch ($name) {
            case "base":
            case "ratelimit":
            case "retry":
            case "test":
            case "timeout":
                return true;
            default:
                return false;
        }
    }
}

File: py/bng2latlongconverter_sdk/features.py
# Bng2latlongConverter SDK feature factory

from bng2latlongconverter_sdk.feature.base_feature import Bng2latlongConverterBaseFeature
from bng2latlongconverter_sdk.feature.ratelimit_feature import Bng2latlongConverterRatelimitFeature
from bng2latlongconverter_sdk.feature.retry_feature import Bng2latlongConverterRetryFeature
from bng2latlongconverter_sdk.feature.test_feature import Bng2latlongConverterTestFeature
from bng2latlongconverter_sdk.feature.timeout_feature import Bng2latlongConverterTimeoutFeature


_FEATURES = {
    "base": lambda: Bng2latlongConverterBaseFeature(),
    "ratelimit": lambda: Bng2latlongConverterRatelimitFeature(),
    "retry": lambda: Bng2latlongConverterRetryFeature(),
    "test": lambda: Bng2latlongConverterTestFeature(),
    "timeout": lambda: Bng2latlongConverterTimeoutFeature(),
}


def _make_feature(name):
    factory = _FEATURES.get(name)
    if factory is not None:
        return factory()
    return _FEATURES["base"]()


# True when this SDK was generated with the named feature class - the
# constructor's tolerance for extend-carried features reads this (an
# active name with no generated class must not become a BaseFeature
# stray when an extend instance carries it).
def _has_feature(name):
    return name in _FEATURES
